app/backend/src/clj/teet/contract/contract_db.clj

```diff
@@ -0,0 +1,154 @@
+     [?c :thk.contract/start-of-work ?start]
+     [(< ?start ?now)]
+     (contract-deadline ?c ?act-dl)
+     [(teet.util.date/dec-days ?act-dl 30) ?deadline-soon]
+     [(< ?now ?deadline-soon)]
+     [(ground :thk.contract.status/in-progress) ?s]]
+    ;Deadline approaching - Days until deadline < 30 deadline is given
+    [(contract-status ?c ?s ?now)
+     [?c :thk.contract/start-of-work ?start]
+     (contract-deadline ?c ?act-dl)
+     [(teet.util.date/dec-days ?act-dl 30) ?deadline-soon]
+     [(> ?act-dl ?now)]
+     [(> ?now ?deadline-soon)]
+     [(ground :thk.contract.status/deadline-approaching) ?s]]
+    ;Deadline overdue - Deadline date has passed
+    [(contract-status ?c ?s ?now)
+     [?c :thk.contract/start-of-work ?start]
+     [(missing? $ ?c :thk.contract/warranty-end-date)]
+     (contract-deadline ?c ?act-dl)
+     [(< ?act-dl ?now)]
+     [(ground :thk.contract.status/deadline-overdue) ?s]]
+    ;Warranty - deadline (and extended deadline if given) has passed and warranty date isn't passed
+    [(contract-status ?c ?s ?now)
+     [?c :thk.contract/start-of-work ?start]
+     [?c :thk.contract/warranty-end-date ?warranty-end]
+     [(< ?now ?warranty-end)]
+     (contract-deadline ?c ?act-dl)
+     [(< ?act-dl ?now)]
+     [(ground :thk.contract.status/warranty) ?s]]
+    ;Completed - deadline and warranty period passed.
+    [(contract-status ?c ?s ?now)
+     [?c :thk.contract/warranty-end-date ?warranty-end]
+     (contract-deadline ?c ?act-dl)
+     [(< ?act-dl ?now)]
+     [(< ?warranty-end ?now)]
+     [(ground :thk.contract.status/completed) ?s]]])
+
+(defn contract-with-status
+  "Used when after datomic query which returns [contract contract-status]"
+  [[contract status]]
+  (assoc contract :thk.contract/status status))
+
+(defn get-contract
+  [db contract-eid]
+  (-> (d/q '[:find (pull ?c [*]) ?status
+             :where
+             (contract-status ?c ?status ?now)
+             :in $ % ?c ?now]
+           db
+           contract-status-rules
+           contract-eid
+           (Date.))
+      first
+      contract-with-status))
+
+(defn parse-target-information
+  [[target project activity]]
+  (let [project-id (:thk.project/id project)
+        activity-id (str (:db/id activity))]
+    {:target target
+     :activity {:activity/manager (user-model/user-name (:activity/manager activity))
+                :activity/name (:activity/name activity)}
+     :project project
+     :target-navigation-info (if (:activity/name target)
+                               {:page :activity
+                                :params {:project project-id
+                                         :activity activity-id}}
+                               {:page :activity-task
+                                :params {:project project-id
+                                         :activity activity-id
+                                         :task (str (:db/id target))}})}))
+
+(defn get-contract-target-information
```

Comment:
`get-` verb prefixes are not necessary, you can just name the function by what it returns.